Composer/packages/lib/shared/src/types/sdk.ts

```diff
@@ -3,15 +3,21 @@
 
 /* eslint-disable @typescript-eslint/no-empty-interface */
 
-interface BaseSchema {
+export interface DesignerData {
+  name?: string;
+  description?: string;
+  id: string;
+}
+
+export interface BaseSchema {
```

Comment:
This was intentionally a private interface. Why do we need to export?

Reply:
Not necessariIy. It's a leak, removed.

Reply:
Mark as resolve after we use template in `shallowCopyAdaptiveAction<T as extends>`